cmd/amp2response.cpp

```diff
@@ -150,12 +152,13 @@ class Accumulator {
     ~Accumulator ()
     {
       // accumulate results from all threads:
+      std::lock_guard<std::mutex> lock (S.mutex);
```

Comment:
no mutex required. Construction and destruction of functors is guaranteed to happen in main thread. Only the [`execute()` method](https://github.com/MRtrix3/mrtrix3/blob/dev/core/algo/threaded_loop.h#L343-L347) is invoked in each thread.